lib/Sema/TypeCheckDeclOverride.cpp

```diff
@@ -928,6 +931,96 @@ static void checkOverrideAccessControl(ValueDecl *baseDecl, ValueDecl *decl,
   }
 }
 
+static GenericSignature *getNewGenericSignature(ValueDecl *base,
+                                                ValueDecl *derived) {
+  auto baseGenericCtx = base->getAsGenericContext();
+  auto derivedGenericCtx = derived->getAsGenericContext();
+  auto &ctx = base->getASTContext();
+
+  if (!baseGenericCtx) {
+    return nullptr;
+  }
+
+  auto baseClass =
+      baseGenericCtx->getInnermostTypeContext()->getSelfClassDecl();
```

Comment:
Since base and derived are always members of a class, I would prefer `baseClass = base->getDeclContext()->getSelfClassDecl()`